src/query/executor/transform/exec.go

```diff
@@ -41,14 +64,26 @@ type simpleOpNode interface {
 // func (n MyNode) Process(queryCtx *models.QueryContext, ID parser.NodeID, b block.Block) error {
 //     return transform.ProcessSimpleBlock(n, n.controller, queryCtx, ID, b)
 // }
-func ProcessSimpleBlock(node simpleOpNode, controller *Controller, queryCtx *models.QueryContext, ID parser.NodeID, b block.Block) error {
-	sp, ctx := opentracingutil.StartSpanFromContext(queryCtx.Ctx, node.Params().OpType())
-	nextBlock, err := node.ProcessBlock(queryCtx.WithContext(ctx), ID, b)
-	sp.Finish()
-	if err != nil {
-		return err
-	}
+func ProcessSimpleBlock(
+	node simpleOpNode,
+	controller *Controller,
+	queryCtx *models.QueryContext,
+	ID parser.NodeID,
+	b block.Block,
+) error {
+	return processBlock(node, controller, queryCtx, ID, b, true)
+}
 
-	defer nextBlock.Close()
-	return controller.Process(queryCtx, nextBlock)
+// ProcessWrapperBlock works similarly to ProcessSimpleBlock, but does not close
```

Comment:
This is probably debatable, but maybe best to just not use the helper in this case? There are already functions which don't (for basically the same reason--they need more control over the block lifecycle). Especially because we comment pretty explicitly down below; might be more clear to just manage the lifecycle directly there.

Reply:
Sounds good, did it this way mostly because I didn't realize that we already had a couple of blocks that don't use it, so happy to revert these changes.